tests/integration/suite/test_node_templates.py

```diff
@@ -47,16 +52,57 @@ def migrated_template_exists(id):
             assert e.error.status == 403
             return False
     id = "cattle-global-nt:nt-" + ns.id + "-" + dynamic_nt["metadata"]["name"]
+    legacy_id = dynamic_nt["metadata"]["name"]
+    legacy_ns = dynamic_nt["metadata"]["namespace"]
+    full_legacy_id = legacy_ns + ":" + legacy_id
+
+    # wait for node template to be migrated
     nt = wait_for(lambda: migrated_template_exists(id), fail_handler=lambda:
                   "failed waiting for node template to migrate")
 
     # assert that config has not been removed from node template
     assert nt.azureConfig["customData"] ==\
         dynamic_nt["azureConfig"]["customData"]
-    wait_for(lambda: admin_client.by_id_node_template(id=ns.name + ":" +
-             node_template_name) is None, fail_handler=lambda:
+
+    def legacy_template_deleted():
+        try:
+            k8s_dynamic_client.get_namespaced_custom_object(
+                "management.cattle.io", "v3", ns.name, 'nodetemplates',
+                legacy_id)
+            return False
+        except ApiException as e:
+            assert e.status == 404
+            print(e)
+            return True
```

Comment:
In place of the True, do a check on the error code, if it's 404 return true, if not let it keep going and time out. Also drop the print